### DISCUSSION TITLE
[AC]vehicle

### DISCUSSION THREAD

**Discussion by dhjska** [trickerer/Trinity-Bots#388] [General]
AC
In Nexus[mapid578], summon the vehicle with [item37815],npcbot will also be on the vehicle, but after a few seconds the vehicle will disappear,
At the final boss, npcbot cannot defeat the boss without a vehicle

npcbot doesn't have this problem at TC terminal, it just flies a little slower, but it's still on the vehicle

[The above content is translated, I can't speak English, first time using github, thanks]

**Comment by trickerer** [comment]
Confirmed, but there is more. At some point AC flying vehicles got broken and they can no longer fly, only jump, for some (stupid of course) reason.

We'll have to wait for that fix for that before new code can be adjusted to support bots and not throw them out.

**Comment by dhjska** [comment]
Thank you for your answer. I saw that you fixed the vehicle problem of ToC5 some time ago and thought that the vehicle was the same, but it turned out that the flight vehicle was different

Thank you for your work

**Comment by trickerer** [comment]
Related: #389

**Comment by dhjska** [comment]
Hi, thank you for fixing this issue

In addition, the repair of delayed attacks is great. Tanks can quickly establish hatred, and long-range DPS will not immediately attract monsters

There are still a few small issues
1. Magebot's Arcano talent does not use arcane missile skills [spell5143]
2. Magebot spell "Mirror imgae" does not work, they will only stand there

translate

**Reply by trickerer** [reply]
1. Can't confirm. Check if spell is disabled for the bot
2. https://github.com/trickerer/AzerothCore-wotlk-with-NPCBots/commit/e3f427546fe8b000f560de9054c864ecb8158991

**Reply by dhjska** [reply]
1.I tested it again
[Arcano]
[Arcane Missiles]
This ability does not work at mage levels 45-63, but can be cast at other levels
2.Thank you

translation

**Comment by dhjska** [comment]
Sorry
I just updated the code today
//Three mages were recruited
The [Arcane]/[Fire]/[Frost] talents are set respectively, and characters are tested from level 10
1.[Arcane]45-63 not used /[Arcane Missiles],10-44,64-80 used
2.[Fire]10-80 can use /[Fireball]
3.[Frost]10-80 not used /[Frostbolt]

4.Mirror ability still does not work, will not attack

translation

**Reply by trickerer** [reply]
https://github.com/trickerer/AzerothCore-wotlk-with-NPCBots/commit/be1b126bc7d212212fc961be4db8878d77d61d67

This fixes 1 and 3.

As for Mirror Image, it works:

![MI](https://user-images.githubusercontent.com/1254989/230756346-5a35cea0-740f-4c68-ac0d-55d0fff6e3fe.jpg)

**Comment by dhjska** [comment]
Updated the code
Now that the mage can cast abilities normally,

The thing about mirrors, since I tested them with [Heroic Training Dummy/entry31146], they just stand there

But it works during a raid

It was my mistake....

translation

**Comment by dhjska** [comment]
Hi, I'm here again 
I found the following problems
1.
mages [Frost]       //spellid44557
Hunter [Survival]  //spellld53290
Cannot trigger effect skill [spell 57669],
I don't know if the code has this effect.
2.
About the paladin
[Retribution]
Divine Storm
Crusader Strike
From what I've seen NPCBOT never use them,
Do not know what the conditions of release are [did not disable them]

translation

**Reply by trickerer** [reply]
I checked the code and now I remember Replenishment-triggering talents were intentionally disabled for some classes.

As for Dinine Storm and Crusader Strike - they were only enabled with 2-handers. This was before you could disable spells - enabled now. https://github.com/trickerer/Trinity-Bots/commit/80f9404891cf6a1126e2fa5eb354d4aeb92962bc

**Reply by dhjska** [reply]
All right. The Five can change careers.

Paladin is usually a single hand + shield, did not notice that this is a two-handed trigger...

After updating the code
The paladin can now use both skills

Thank you

translation

**Comment by dhjska** [comment]
In 25HICC
After boss_the_lich_king releases skill [74297],
NPCBOT is not teleported to the In Frostmourne Room with the player

Test on TC server that they can be delivered In Frostmourne Room

Can npcbot detect and dodge boss_the_lich_king's AOE skills
It s obvious on boss_lady_deathwhisper

I don't know if I can articulate that.  English is so bad :)
translation

**Reply by trickerer** [reply]
I know of Frostmourne Room thing. In many cases, additional 25-man boss mechanics aren't supported by bots since it's extremely difficult to do with bots anyways, you need at least 2-3 people for that.

AFAIK bots do avoid Death and Decay on Deathwhisper, but they don't do that immediately and take some big hits.

**Reply by dhjska** [reply]
Yes, solo is hard. So I made team Buff enhancements and tested them
Now I want npcbot to be able to transport as well as the TC side.
npcbot does not transmit
Lich King will become huge and damage increased by 10000%....

translation

**Comment by dhjska** [comment]
@trickerer 
After updating
The [sendto] command and subcommands are invalid

translation

**Reply by trickerer** [reply]
This isn't a chat for random problem reports. If you've encountered a problem report is in issues. Fill the template and provide all the additional info you can, otherwise chances that your problem will be fixed drop drastically.

**Reply by dhjska** [reply]
Sorry, I'm not familiar with Github and my English is not good
So I dare not click on other places indiscriminately...

translation
